src/components/Form/Integer.js

```diff
@@ -0,0 +1,130 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { FormControl } from 'patternfly-react';
+import { eventValueHandler } from './utils';
+import { getSequence, setNativeValue } from '../../utils/utils';
+import { isMinus, KEY_CODES, INPUT_NAVIGATION_KEYS } from '../../constants/keys';
+
+const NON_NEGATIVE_INTEGER_KEYS = [
+  ...INPUT_NAVIGATION_KEYS,
+  ...getSequence(KEY_CODES[0], KEY_CODES[9]),
+  ...getSequence(KEY_CODES.NUMPAD[0], KEY_CODES.NUMPAD[9])
+];
+
+const INTEGER_KEYS = [...NON_NEGATIVE_INTEGER_KEYS, KEY_CODES.MINUS, KEY_CODES.HYPHEN_MINUS, KEY_CODES.NUMPAD.SUBTRACT];
+
+const POSITIVE_INT_REGEX = /^[1-9]\d*$/;
+const NON_NEGATIVE_INT_REGEX = /^(0|[1-9]\d*)$/;
+const INT_REGEX = /^(-?0|-?[1-9]\d*)$/;
+
+const PRECEEDING_ZEROS_POSITIVE_INT_REGEX = /^0*([1-9]\d*)$/;
+const PRECEEDING_ZEROS_INT_REGEX = /^(-?)0*([1-9]\d*)$/;
+
+const fixPrecedingZerosPositiveInt = value => {
+  const match = PRECEEDING_ZEROS_POSITIVE_INT_REGEX.exec(value);
+  return match && match.length === 2 ? match[1] : '';
+};
+
+const fixPrecedingZerosNonNegativeInt = value => {
+  const match = PRECEEDING_ZEROS_POSITIVE_INT_REGEX.exec(value);
+  return match && match.length === 2 ? match[1] : '0';
+};
+
+const fixInt = (value, oldValue, keyCode) => {
+  if (value.length === 0 && isMinus(keyCode)) {
+    if (oldValue) {
+      return oldValue < 0 || oldValue === '-0' ? oldValue : `-${oldValue}`;
+    }
+    return '-0';
+  }
+
+  const match = PRECEEDING_ZEROS_INT_REGEX.exec(value);
+  if (match && match.length === 3) {
+    if (match[1]) {
+      return `${match[1]}${match[2]}`;
+    }
+    return match[2];
+  }
+  return '0';
+};
+
+export const Integer = ({ id, value, defaultValue, onChange, onBlur, positive, nonNegative }) => {
+  let allowedKeys;
+  let validRegex;
+  let fixAfterValue;
+  let min;
+
+  if (positive) {
+    allowedKeys = NON_NEGATIVE_INTEGER_KEYS;
```

Comment:
can i ask the original source of these functions? I am just noting that there is some additional validations here that may not be present in other libraries (so it would be nice to know the distinctions made). For example, [Formik](https://github.com/jaredpalmer/formik/blob/master/src/utils.ts#L98) has very simple assertion functions. Does it make sense to self implement these or use a 3rd party? I'm fine w/ this, but would just like to know the API surface is well supported.

Reply:
My primary concern w/ form validations / form HOCs such as these is that they will:

1. Be flexible to different variation
2. Will support local state management. Many times in the past I have the feeling that redux-form is creating an unnecessary burden, and that form state should be locally managed (like you have done here).

Reply:
I started from some SO question like this [one](https://stackoverflow.com/questions/5535449/javascript-keycode-allow-number-and-plus-symbol-only) (don't remember which exact one was it) and wrote it into the current form. 

I looked at Formik but it doesn't have anything for this use-case. 
- I wanted to have a field which would allow user to press only the valid keys without the need to show errors (i.e. positive integer without 0). 
- We could use formik components and validation lifecycle but I am not sure if it is worth it. Formik doesn't help with the validation itself.

I could write the component in more generic way. There was a suggestion to submit it to pf-react. Do you think that is a good idea?

btw: you can try the component [here](https://clever-allen-639701.netlify.com/?component=Form%2FG&fixture=Integer&editor=true)

Reply:
I agree. I will start looking into the table first then.